ecllibrary/std/Uni.ecl

```diff
@@ -398,4 +398,17 @@ EXPORT unicode GetNthWord(unicode text, unsigned4 n, varstring localename = '')
 EXPORT ExcludeNthWord(unicode text, unsigned4 n, varstring localename = '') :=
     lib_unicodelib.UnicodeLib.UnicodeLocaleExcludeNthWord(text, n, localename);
 
+/**
+ * Returns the source string with the all characters that match characters in the search string replaced
+ * with the character at the corresponding position in the replacement string.
```

Comment:
I would guess that it depends on the Character Iterator that you are using.  You may need to wrap a try/catch on some (or all) of the operations if it throws an exception for an ill-formed UTF-16 string.  Do you already know or should I study the iterator?